src/app/controllers/submissions.server.controller.js

```diff
@@ -226,51 +221,27 @@ async function sendOneEmailAutoReply(
   index,
 ) {
   const { form, submission } = req
-  const submissionId = submission.id
-  const defaultSubject = 'Thank you for submitting ' + form.title
-  const defaultSender = form.admin.agency.fullName
   const defaultBody = `Dear Sir or Madam,\n\nThank you for submitting this form.\n\nRegards,\n${form.admin.agency.fullName}`
   const autoReplyBody = (autoReplyEmail.body || defaultBody).split('\n')
 
   // Only include the form response if the flag is set
   const templateData = autoReplyEmail.includeFormSummary
     ? { autoReplyBody, ...renderData }
     : { autoReplyBody }
-  let autoReplyHtml
   try {
-    autoReplyHtml = await renderPromise(
+    const autoReplyHtml = await renderPromise(
       res,
       'templates/submit-form-autoreply',
       templateData,
     )
-  } catch (err) {
-    logger.warn('Render autoreply error', err)
-    return Promise.reject(err)
-  }
-  const senderName = autoReplyEmail.sender || defaultSender
-  // Sender's name appearing after ( symbol gets truncated. Escaping it solves the problem.
-  const escapedSenderName = senderName.replace('(', '\\(')
 
-  const mail = {
-    to: autoReplyEmail.email,
-    from: `${escapedSenderName} <${config.mail.mailFrom}>`,
-    subject: autoReplyEmail.subject || defaultSubject,
-    // Only send attachments if the admin has the box checked for email field
-    attachments: autoReplyEmail.includeFormSummary ? attachments : [],
-    html: autoReplyHtml,
-    headers: {
-      [EMAIL_HEADERS.formId]: String(form._id),
-      [EMAIL_HEADERS.submissionId]: submission.id,
-      [EMAIL_HEADERS.emailType]: EMAIL_TYPES.emailConfirmation,
-    },
-  }
-  try {
-    return sendNodeMail({
-      mail,
-      options: {
-        retryCount: config.mail.retry.maxRetryCount,
-        mailId: `${submissionId}-${index}`,
-      },
+    return MailService.sendAutoReplyEmail({
+      form,
+      submission,
+      attachments,
+      index,
```

Comment:
Similarly for this bit - generation of HTML can be thought of as an internal detail for sending the autoreply email.

Reply:
Requires `res.app` to be passed into the service to get the rendering engine to be able to generate those dynamic html. Do you have any suggestions as to how the function can remain pure?

Reply:
how about using the [ejs](https://www.npmjs.com/package/ejs) engine directly via import? I think the roundabout way that we're using to obtain a reference to the engine is an anti-pattern; Express needs it to render templated pages via `res.render`, but that is not what we're doing here.

Reply:
I didn't know we could do that. That would definitely make things a lot simpler. Thanks!